pkg/util/ec2/ec2.go

```diff
@@ -96,14 +96,26 @@ func GetSourceName() string {
 var instanceIDFetcher = cachedfetch.Fetcher{
 	Name: "EC2 InstanceID",
 	Attempt: func(ctx context.Context) (interface{}, error) {
-		return getMetadataItemWithMaxLength(ctx, imdsInstanceID, false)
+		hostname, err := getMetadataItemWithMaxLength(ctx, imdsInstanceID, UseIMDSv2(false, false))
+		if err != nil && pkgconfigsetup.Datadog().GetBool("ec2_imdsv2_transition") {
```

Comment:
> can we put the flag name in a const as well?

done

> I think we want to see the value of the bool config flag in a separate debug/trace log statement before this condition otherwise we won't know if an error made us skip pulling the DMI value or the lack of an error when debugging.

The error seems to be logged here: https://github.com/DataDog/datadog-agent/blob/3f972c1d3b36d56bc34a8732483644f95363041b/pkg/util/hostname/providers.go#L205

I just split the if condition in 2 if conditions to only have the DMI log when the flag is set to true, I will let you check to see if it fits to you